### PR TITLE
add notifyStop() for metaClient

### PR DESCRIPTION
**Pull request by liuyu85cn**
#### What type of PR is this?
- [X ] bug
- [ ] feature
- [ ] enhancement

#### What problem(s) does this PR solve?
Issue(s) number: 

Description:

It is reported if not "add host" metaClient will stuck at "waitForMetaready()" and can't stop. 
This pr try to fix this. 

#### How do you solve it?

waitForMetaready() will detect a flag in each loop, before retry. Set this flag. 
  
#### Special notes for your reviewer, ex. impact of this fix, design document, etc:

Because no UT call metaClient.notifyStop() before, and it is hard to know which UT will affect by this, just run CI 

#### Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

#### Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

**Review by yixinglu — APPROVED** [review]

**Comment by HarrisChu** [comment]
fix https://github.com/vesoft-inc/nebula/issues/3476

**Review by critical27 — APPROVED** [review]
👍 , quick fix

**Review by panda-sheep — APPROVED** [review]
Good job！

File: src/storage/StorageServer.cpp [hunk 1]
```diff
@@ -363,6 +363,9 @@ void StorageServer::notifyStop() {
     serverStatus_ = STATUS_STOPPED;
     cvStop_.notify_one();
   }
+  if (metaClient_) {
+    metaClient_->notifyStop();
+  }
```
Comment (panda-sheep) [inline src/storage/StorageServer.cpp:368]: 👍

**Pull request merged**